### DISCUSSION TITLE
Front end focado em UI / UX

### DISCUSSION THREAD

**Discussion by Cleysonlb** [frontendbr/forum#415] [Discussão]
Fala galera beleza?! Gostaria de trazer essa discussão, sobre front end voltado pro UI UX, como é o mercado pra esse cara?! É necessário esse cara ser fullstack ?! Alguém atua como front end UI UX e pode dividir um pouco do dia a dia e da experiência?

**Comment by laurindo** [comment]
@cleilson, não é obrigatório esse cara ser Full-Stack. Mas ele ter uma base
legal em relação a UI /UX e a parte de programação, seja frontend, backend
ou full, é com certeza um perfil diferenciado.

Atualmente na empresa em que trabalho, a necessidade é encontrar um perfil
desses  e muitos desenvolvedores do nosso time estão estudando pra agregar
esse conhecimento.

Vejo também que lá fora é um requisito diferencial é também é uma área que
está em alta também atualmente.

Resumindo, você ser bom em UX/UI + frontend ou backend, com certeza é um
plus diferencial.

**Comment by laurindo** [comment]
Desculpa, é  @cleyson.

**Comment by woliveiras** [comment]
Galera, dos Front End UI que aparecerem, eu estou precisando de ajuda pra montar a sessão de UI do Carreira Front End: https://github.com/woliveiras/front-end-career/issues/20

Se alguém puder contribuir vai ganhar meu abraço virtual!

**Comment by felipefialho** [comment]
Só linkando

#32

**Comment by dionarodrigues** [comment]
Bom, minha carreira começou como designer - me tornando web designer - e hoje trabalho como front-end UI/UX.

Ainda hoje, trabalhar como web designer é algo bem interessante quando se busca uma empresa pequena (a parte chata é que na maioria das vezes o salário também é pequeno). Contudo, ser web designer em uma startup pode ser uma genial ideia.

Enquanto web designer, procurei refinar muito o design dos layotus que eu criava. Estudei bastante e pratiquei também, afinal, essa é uma característica importante de um designer. Porém, junto com esse refinamento no design procurei me tornar um expert em HTML, CSS e JS (leia-se jQuery). Estudei um pouco de banco de dados, houve uma época em que eu me interessei muito por SEO, acessibilidade e aprendi o funcionamento do WordPress - e a criar temas para ele do zero. 


Esse planejamento me pareceu ótimo até eu ficar desempregado e começar a procurar vagas como web designer. Na época - e acredito que até hoje - ser web designer não se ganha lá aquelas coisas quando comparado com um front-end. E desde então passei a ser um front-end UI/UX. 

Claro! Dinheiro é importante. No entanto eu também adoro front-end. Gosto de ver o resultado dos meus trabalhos, de aperfeiçoar a UX e mudar a vida das pessoas. Pensei: posso continuar lidando com o design de certa maneira; não estarei jogando fora todos os anos que estudei para melhorar o meu design. 

Mas aí tive outra grande barreira: a forte diferença no mercado entre um front-end UI e um front-end engineer.

Percebi que o mercado busca, em grande parte, por Front-end engineer. Um profissional mais voltado pra lógica, com algum conhecimento de back-end. E isso fica muito evidente quando se olha as vagas de emprego. Quando se vai fazer testes para as vagas, quando se vai nas entrevistas de emprego.

Então, penso que é interessante focar em algumas tecnologias que o mercado exige - aquelas em que nos sentimos confortáveis - e nos aperfeiçoar nelas. Hoje, apesar de estar sempre focado em front-end UI, busco aperfeiçoar bastante meus conhecimentos em JS e estar a par de Angular, React e demais modas do mercado.

Não penso que um front-end precisa ter um perfil full-stack, a não ser que ele realmente consiga se aprofundar em todos os temas. Mas é preciso ser bom naquilo que ele se propôs a ser.

**Comment by rubenmarcus** [comment]
Existe futuro pra quem quiser só se focar nisso? Estou estudando https://www.coursera.org/specializations/interaction-design

Esses cursos da Universidade da California e estou gostando muito..

Sei do mantra Javascript,Javascript,Javascript, e eu gsto muito de Javascript, do ecossistema NPM e do Node, Mas eu me vejo muito mais mexendo com UI/UX e pra mim javascript sempre foi encarado como algo a se fazer porque é necessário e não porque é meu foco.

Pra falar a verdade eu venho duma epoca (2000) em que existia os Webmasters e Webdesigners, e eu era os dois. Até hoje não vejo muita diferença, vejo que se separou um pouco as funções e tem especializações e especificações novas.

Mas eu pirava mesmo em ver os layouts nos jpegs e psds e eles no ar. A parte de desenvolver funcionalidades nunca foi o que mais pirei. Acho muito louco também, Mas meu foco era esse, fazer um design no Photoshop e ver ele aplicado no browser.

Existe caminho pra uma boa carreira pra quem gosta disso? Porque eu vivi um dilema todos esses anos, de: 

- meu pai me falando pra eu focar em back-end e Javascript tambem, fiz curso de C# MVC, de PHP , etc etc, e até aprendi um pouco, mas eu não aplicava no dia-dia e então muita coisa eu perdi. O conceito digamos que não, mas a pratica e malicia de como fazer sim. Pra mim é como se eu visse um código eu sei o que está sendo feito ali, sei meio que "onde" mexer, mas pegar do zero, sem orientação de quem fez é uma missão quase impossivel pra mim

- Você é bom com design, mas não é webdesigner, é um recortador de layouts e implementador de temas.

Pra mim isso sempre foi muito ruim. Eu me vejo muito nos próximos anos com um portfolio com cases de Design/Identidade Visual e UI/UX bem legais, assim como coisas de código que envolvam principalmente coisas visuais (animações e etc) , do que um cara que desenvolveu funcionalidades para um sistema.

Existe mercado pra isso? Aqui e lá fora?

Porquê tem vagas me procurando de fora,vendo que tenho experiência com angular e anos de profissão. mas minha experiência nunca foi além de mexer com interações na tela, e ai chega na etapa de mostrar lógica e algumas vezes consigo, outras me embanano, então eu estou num ponto de pensar se é isso que quero mesmo


Eu, vejo um caminho a seguir é desenvolver esse ano ainda no primeiro trimestre o melhor portfolio  que conseguir ,junto com temas free de wordpress, e também com códigos pra auxiliar outras pessoas coisas free ( tipo tela de login e etc), e do meio pro ano pra frente eu ia entrar num curso de JS aqui c node e só fazer isso tambem..

Não me vejo fullstack, gosto muito da parte visual, Mas.. odeio depender dos outros também então acho que essa parte de JS que mexe com back também preciso aprender e me aprofundar, ao invés de só ficar lendo sobre e nunca por a mão na massa.

**Comment by rubenmarcus** [comment]
@diogorodrigues Então mano mas o Behance e o Dribble é pros UX/Designer o que o Git é pros coders. Não tem mercado pra quem gosta de UX? 

Eu me vejo com um perfil bem parecido com o seu, mas, fui "forçado" a mexer com código, na verdade eu gosto muito ,mas 90% dos projetos que trabalhei era do perfil "se vira", então aprendi na raça no dia-dia.
Mas não aprendi o quanto precisaria ou gostaria.

Então eu to pensando se foco muito na parte visual e vou mais pra isso, ou se foco na parte de back que é bem meu fraco, e que eu gosto também ,mas não é o que eu gosto mais..

Se você for pegar um cara que fez Ciências da Computação ou Análise de Sistemas e ai, comparar o código dele com o meu, a velocidade de implementação e a qualidade do código dele provavelmente é melhor.
Agora se você for comparar implementação front de tela, ou criação visual, provavelmente o cara vai fazer algo nível Paint ou vai usar algum framework pronto, Quando eu me sinto totalmente a vontade pra fazer o que for..

Só que ai por exemplo:

Aqui na Printi eu fui contratado como front.. o clássico pra implementação de telas e interações visuais no site, como também design responsivo. Mas o UX e o Design quem fez foi uma agência. E eu tenho knowhow de fazer também. Mas não fui contratado pra isso. O back-end e o envio de dados via Angular e PHP foram os back-ends que fizeram. Já nessa parte provavelmente se fosse eu fazendo eu ia usar coisas prontas e quase nada ia fazer do zero.

Esse é meu perfil, e eu quero mudar... Nem é pra ganhar mais tanto o foco, e sim pra ter mais independencia e não me sentir tão amarrado entre um e outro

**Comment by rubenmarcus** [comment]
Relacionado : 
Why do so many UI developers think backend will go away?
https://www.quora.com/Why-do-so-many-UI-developers-think-backend-will-go-away

**Comment by edkf** [comment]
Legal, meu perfil é bem parecido com o do @laurindo e do @rubenmarcus.
Trabalhei durante um tempo em agências como designer e agora sou front-end em um estúdio de design.

Comecei a estudar html/css/js quando ainda era designer, pois acreditava que se eu entendesse como as coisas funcionavam “por trás”, tomaria decisões mais coerentes na hora da criação de interações. Acredito um pouco nisso até hoje. 🙂

Depois de um tempo acabei tendo a oportunidade de ir para uma startup ser o Product Designer deles, o que foi muito legal também. Lá consegui aplicar um pouco de UX no produto e aprender bastante código. A vantagem de trabalhar em times enxutos é que você acaba aprendendo de tudo um pouco.

Acabei saindo da startup depois de 2 anos e comecei a trabalhar full-time como front-end.
O legal disso tudo, é que a Lead Developer que me contratou na época (@nathaliabruno), disse que no processo seletivo encontrou pessoas que tinham muito mais experiência como desenvolvedor que eu, porém como já havia trabalhado como designer, teria bastante cuidado visual na hora de montar as telas de projetos futuros. E isso para um estúdio de design era algo bem relevante.

Particularmente, acredito que o desenvolvedor front-end ter a responsabilidade de realizar pesquisas de UX, desenvolver a parte visual de um projeto e escrever todo o código front-end, acaba o deixando um pouco sobrecarregado. Porém alguns cuidados ele pode tomar, por exemplo:

* Ter noção de que o desenvolvedor também tem um papel muito importante na hora de criar uma experiência agradável para o usuário. Assim como todas as pessoas do time (Vendas, suporte etc).
* Entender as decisões que o designer tomou na hora da criação das interfaces.
* Ter uma noção básica de tipografia, hierarquia de informação e psicologia das cores. Quando se sabe um pouco disso, você pode resolver problemas de interface simples sem um designer.
* Conhecer a persona que usa o produto na qual você trabalha, e entender o contexto de uso do produto.

Essas são apenas algumas características que eu acredito que são legais de um front-end ter, mas gostaria de ouvir a opinião de vocês também. 🙃

**Comment by laurindo** [comment]
Um artigo bacana que fala um pouco de UX/UI aplicado em um produto de
sucesso.

https://medium.com/@milfont/o-que-podemos-aprender-com-o-snapchat-8057aff67a17#.bj77ty6kp

**Comment by Cleysonlb** [comment]
@edkf assisti uma palestra da Nathalia no front end carioca do ano passado e foi ótimo!
Obrigado por dividir sua experiência, eu sou front end mas sempre puxei mais pro lado do design e de tentar criar uma boa experiencia pro cara que ta usando, atualmente estou estudando mais sobre UX e aplicando conceitos na empresa onde trabalho.

**Comment by cmpereirasi** [comment]
## **Uma discussão interessante** 🎉
Vai depender muito da empresa e também do profissional, porque algumas empresas pelo que eu andei pesquisando e observando. Contratam um cara que vai cuidar especificamente da **Experiência do Usuário** , porém que tenha um conhecimento solido em: **HTML, CSS e JS**. E também percebe que a galera que está entrando para o mundo de **UX/UI**, são pessoas que já trabalham com **Design Gráfico** e áreas relacionado ao **Design** e para quem já trabalha como **Front-End** acaba sendo uma obrigação ter conhecimento solido em UX.
### Agora vamos fazer uma analogia simples

- Front-End com conhecimento em UX e UI, concorrendo uma vaga, é claro que ele sai na frente ⚔🎉

- UX Design com o conhecimento em Front-End e UI, obviamente será um profissional de peso 😎 📝

- UI Design com conhecimento em Front-End e UX, é claro que será um peso concorrendo as vagas ✏️

**Comment by rubenmarcus** [comment]
@cmpereirasi Sim, mas em questão de vagas/mercado, as vagas com foco, ou até ver se tem a palavra UI/UX comparando com as vagas que focam em palavras como Engineer,Angular2,React,Vue,Node.js, além do gap salarial, um Front-end Engineer Senior e um UI/UX Front-end Senior, o salário deve ser bem diferente ou não??

Outra coisa,um problema que encontramos aqui na Printi, e que me disseram que me contrataram foi:

- Eles procuravam um front que fizesse as telas novas do site ( O site que está no ar hoje ), e que não precisasse ser um expert em JS, mas que tivesse conhecimento necessário para se virar,e também experiência suficiente pra codar as telas novas,sem precisar de ajuda de ninguém. Eles falaram que de 20 candidatos, eu fui o único que eles viram que tinha experiência para isso. A maioria eram designers, que mal tinham experiências, ou que tinham muito cursos online e nenhum projeto em produção.

Outra coisa, Aqui usamos Angular 1, e eu não fiz nem 1% do código em Angular, que foi feito por um back-end PHP.. eles estão buscando um front agora mais voltado a JS, coisa que pra mim é muito interessante aprofundar, mas é complicado porquê comecei esse curso de UX e pretendo terminar nesse primeiro semestre, e outra meu tempo aqui é bem restrito,porquê tem muita implementação,sobra quase nada pra eu sentar do lado dos back-ends e falar : quero participar da implementação das funcionalidades também.

Nem os modais que estão em Angular ,eu que mexo, pois o cara fez aqui, e pra entender a lógica dele, levaria mais tempo do que ele fazer. 

Então assim é minha opinião, pra um profissional ser focado só em UX ou só em JS, ele tem que ser muito BOM nisso, e ter clientes fiéis já, ou estar em uma empresa estruturada.

No meu caso eu estou numa empresa que permite ser bom nos dois, e foi isso que eu falei na entrevista aqui, queria ser senior em front-ux dentro da empresa e deixar de ser junior em js e back-end e passar a ser pleno e ter mais autonomia.. é ruim você não implementar funcionalidades. Pelo menos eu me sinto mal, mesmo sabendo que a implementação de telas é algo bem importante, e que eu vejo milhares de usuários usando as telas que eu codei, dentro da Printi e fora também , mas hoje eu sinto isso, meu coração fala pra eu focar muito em UX e na parte visual/gráfica, mas minha cabeça fala pra eu cair de cabeça em JS, porquê é o que mais vejo pedir nas vagas,e que eu tenho certeza que faria um grande diferencial na minha carreira

**Comment by danvitoriano** [comment]
ótima issue.

**Comment by renanrboliveira** [comment]
A minha experiência é a seguinte:
Trabalhei um bom tempo como backend(Java) e lá fazíamos de tudo um pouco, mas depois de um tempo comecei a me destacar no frontend, assim os meus colegas de trabalho viam tirar dúvidas de javascript e css comigo, então fui focando mais nessa área. Depois procurei uma oportunidade pra trabalhar só como frontend e a partir de um freela, hoje estou em uma empresa fulltime como frontend.

Até então não tinha ouvido sobre "_frontend engineer_", é um termo novo pra mim, mas é onde exatamente me identifico, consigo desenvolver um site normalmente, mas meu foco principal é JS, gosto muito de interagir com ele, no qual desenvolver uma aplicação toda no browser requer muito javascript e eu estava fazendo um projeto que era em Vuejs. 

Às vezes me pego pensando e tendo muitas dúvidas em relação a área de frontend, pois não sou expert em design e nem UI/UX, e prefiro ter um design gráfico que me dê um layout e eu transfira isso pra o HTML, CSS e JS. O que realmente preciso aprender nessa área, já que não sei desenhar lindos sites.

Mas vi nos comentários de você que dá pra ser um bom frontend engineer com essa deficiência(Não ser um bom design), e isso vai partir muito das necessidades das empresas.

Atualmente meus estudos estão voltados apenas pra JS e Frameworks, mas vou começar a focar em testes de usabilidades(Vou fazer um curso mês que vem) e explorar mais UI/UX.

No mais, muito bons os comentários, no qual está me ajudando.

**Comment by cmpereirasi** [comment]
@rubenmarcus, exatamente! Até citei que varia de empresa para empresa e do profissional também. Por exemplo na gringa, vejo muita vaga de **Especialista**👔  em  **UX/UI** e **Front-End** por aí vai... Eu tenho visado o mercado **Canadense**🇨🇦. Pretendo **migrar** futuramente e para melhorar o meu **SKILL**📚, uso [FRESHGIGS](https://www.freshgigs.ca/jobs) para procurar as vagas relacionado e ter noção do que as empresas estão buscando. 👊🏿

**Comment by rubenmarcus** [comment]
cara vejo muito esse Jooble tem vagas no mundo todo, já procurei até na India,Paquistão,Hong Kong, Tailandia,China que são os paises que mais gosto (Ásia),vou te falar que se pudesse eu trabalharia só com freelas ou remoto viajando a cada 3 meses em algum pais, seja Africa, Europa,Ásia, não tenho um preferido não , viajaria em todos os paises que conseguisse.

**Comment by eubond** [comment]
Muito bacana o relato da galera, ótima thread. Mas, eu fiquei pensando então, pelas comparações. O cara de front, focado mais em UX não costuma existir. O que existe é o designer UX, com experiência em front?

Eu, diferente um pouco da galera, migrei dá área de motion, principalmente flash. Sendo assim, essa parte de experiência do usuário, com relação a animação principalmente, era parte do dia a dia. Agora no front, eu tava querendo, agregar as 2 coisas, o front com o background de motion. Inicialmente o que pensei, foram nas tecnologias nesse sentido, porém, agora lendo a thread, comecei a pensar em como utilizar esse bg no sentido de interação do usuário e não só animação. O que vocês acham que faz mais sentido, como profissional de front, tecnologias nesse sentido, ou UX? Claro, não esquecendo de tudo que é importante e básico para a nossa função.

Desculpem, se desviei um pouco a thread, é que surgiu o insight a partir dela.

Valeu.

**Comment by rubenmarcus** [comment]
@eubond tem a parte de jogos também. É uma carreira interessante. Desenvolver games para web

**Comment by eubond** [comment]
@rubenmarcus Pois é mano, por coincidência, foi daí que veio a minha experiência com motion, eu trampava com jogos web. Minha formação é de disso na vdd. E também por isso, pensei em ir pro lado da animação, mas como disse, vendo a thread, fiquei "confuso". Apesar de gostar da área, eu gosto mais do foco em animação do que em games mesmo, principalmente pra programar, full.

Então, to pedindo uma pitada de experiência de vcs. ;)

**Comment by simaodeveloper** [comment]
Canal focado no assunto [UXNOW](https://www.youtube.com/channel/UCgfaifzmqadwKyCd0lagylQ)

**Comment by lucianobarauna** [comment]
Caaaarraa. Que discussão sensacional @Cleysonlb, ainda bem que eu pesquisei antes de começar essa dúvida aqui. Me identifico com uma galera do post só que meu caso foi diferente eu acho rs.

Meados de 2011 eu tinha acabado de terminar um cursinho de webdesigner com isso entrei para uma empresa só na responsabilidade de ficar cuidando de html, css e design sendo esse só manutenção de algumas coisas.

Depois de um tempo e já com um background bom de html e css resolvi cursar a faculdade que foi Design justamente por conta de agregar mais ao meu "front" só que chegou um momento que eu tive que aprender o javascript para conseguir fazer as coisas mágicas e isso foi me puxando mais e mais para esse lado da interação, comportamento e soluções que facilitam a vida do usuário.

Hoje estou focado em aprender JS para conseguir fazer essas maravilhas além de se capacitar para concorrer as vagas de emprego do mercado, isso é uma coisas que me deixava muito preocupado as vagas de front-end só pedem a parte de graduação em **Cinciência, Sistema ou afins** mais como nossa área não vê muito isso então fico um pouco mais tranquilo rs ás vezes e ter design sabendo código é um diferencial isso foi sempre o que eu ouvi mais sempre fico com o pé atrás por conta da mentalidade de umas pessoas.

Mais hoje depois de 6 anos nessa empresa resolvi sair dela para me aprofundar em JS numa outra oportunidade que foi conseguida por conta desse meu background de Design rsrs

Num total ás vezes você pode ficar dividido do que eu sou (eu fico assim rs) mais no fundo por mais dessas vagas de front terem isso acho um plus vc ser formado em Design saber codar e o importante é mostrar o que vc sabe fazer.

**Comment by danilowoz** [comment]
**Acredito** que pode-se dividir em dois tipos de profissionais:

### Front-end com background de design (UI/UX)
Muitas vezes vai ser o profissional que não precisará de layout para montar partes (eu disse partes) da interface e vejo por experiência própria que é um dos profissionais mais solicitados, pois conseguem diminuir muitas etapas (concepção, layout, aprovação, viabilidade e desenvolvimento), o que tornará muitos processos mais ágeis, mas isso claro vai depender do tipo de trabalho e fluxo dentro do time. Além disso, esse cara estará sempre pensando em pontos de usabilidade, acessibilidade e principalmente em manter a concistência de toda interface. 

### Designer com conhecimento de Front-end
São os caras que tem como _core_ UI/UX e possuem o diferencial de criar interfaces pensando no esforço necessário e viabilidade técnica, o que evita muitos gargalos na hora de desenvolver e supresas no final do projeto. Sem falar que poderá criar o seu próprio protótipo e apresentações para aprovação etc... 

Hoje em dia vejo muitos designer querendo aprender HTML/CSS/JS e acho isso fantástico, pois acaba com aquela história de "Converse mais com o seu designer", pois ele deixa de ser "somente" o designer que te ferra a vida e sim o cara que vai te ajudar a implementar, tirar do .psd ou .sketch será algo natural e não cheio de complicações e gambiarras. 

Tenho a história de um amigo meu designer / UX que foi tentar a vida em Berlim, porém estava com dificuldade para encontrar uma vaga, pois se tivesse o Mínimo de HTML/CSS, estaria empregado faz tempo. Bem loco!

#### Experiência própria
Sou formado em design, mas dentro da primeira agência que trabalhei foi algo natural focar em desenvolvimento, pois tinha o olhar refinado para as interfaces, animações, interações e etc, tanto que hoje é 80% do meu trabalho é codificar, MAS isso não quer dizer que deixo de ser designer, pois Design é resolver problemas e não somente "desenhar" interfaces. Sendo assim continuo trabalhando para construir a melhor experiência para o usuário, facilitar ele a encontrar o seu objetivo e manter a experiência projeto coerente. 

Thanks a lot, guys! :smile:

**Comment by Cleysonlb** [comment]
Obrigado a todos que deram suas opiniões e falaram sobre suas próprias experiências <3 Vocês são de mais!

**Comment by jpcmf** [comment]
Pessoal, um pouco atrasado, mas muito legal esse thread! Muito bom conhecer a perspectiva de vocês sobre o assunto e conhecer a experiência de cada um que escreveu. Sou formado em Design Gráfico e também tenho essas dúvidas e anseios. Sempre trabalhei com Web Design (HTML/CSS/jQuery) e atualmente tenho estudado muito JavaScript (ECMA5) mas ainda não decidi qual framework (Angular, Vue ou React) estudar. Gosto muito de ver aquele layout foda ganhando vida e isso que me faz continuar nessa profissão. Abraço a todos!

**Comment by marcos-jesus** [comment]
Muito bom, tenho 1 ano como Desenvolvedor Frontend estou louco para aprender sobre UX / UI e agregar conhecimento a minha carreira de Front.

Quero ser um Frontend Engineer, muito bom em lógica, voltado a soluções simples e brilhantes, com grandes aplicações sem gargalos com uma perfomance escalavel.

E ter um conhecimento legal de UX/UI saber fazer prototipação de telas antes de começar a codar as telas aleatórias que é como eu faço atualmente, zero planejamento não quero isso para minha carreira, quero algo organizado e de execelência.
